tei_adapter/tei_id_extractor/src/main/scala/weco/catalogue/tei/id_extractor/PathIdManager.scala

```diff
@@ -0,0 +1,182 @@
+package weco.catalogue.tei.id_extractor
+
+import scalikejdbc.TxBoundary.Try._
+import scalikejdbc._
+import uk.ac.wellcome.json.JsonUtil._
+import uk.ac.wellcome.messaging.MessageSender
+import uk.ac.wellcome.storage.s3.S3ObjectLocation
+import uk.ac.wellcome.storage.store.Writable
+import weco.catalogue.tei.id_extractor.PathIdManager._
+import weco.catalogue.tei.id_extractor.models.{
+  TeiIdChangeMessage,
+  TeiIdDeletedMessage,
+  TeiIdMessage
+}
+
+import java.time.Instant
+import scala.util.{Success, Try}
+
+class PathIdManager[Dest](pathIds: PathIdTable,
+                          store: Writable[S3ObjectLocation, String],
+                          messageSender: MessageSender[Dest],
+                          bucket: String) {
+
+  def handlePathChanged(pathId: PathId, blobContent: String): Try[Unit] = {
+    DB localTx { implicit session =>
+      for {
+        idRow <- selectById(pathIds, pathId.id)
+        pathRow <- selectByPath(pathIds, pathId.path)
+
+        _ <- (idRow, pathRow) match {
+          case (Some(row), None) =>
+            update(pathId, blobContent, row)
+          case (None, None) =>
+            insert(pathId, blobContent)
+          case (None, Some(row)) =>
+            deleteOldIdInsertNewId(pathId, blobContent, row)
+          case (Some(idRow), Some(pathRow)) =>
+            if (idRow == pathRow) {
+              update(pathId, blobContent, idRow)
+            } else {
+              deleteOldIdUpdateNewId(pathId, blobContent, idRow, pathRow)
+            }
```

Comment:
Minor: I would roll the `if` statement into the pattern match to keep it all at a consistent indentation level.

```suggestion
          case (Some(idRow), Some(pathRow)) if idRow == pathRow =>
            update(pathId, blobContent, idRow)
          case (Some(idRow), Some(pathRow)) =>
            deleteOldIdUpdateNewId(pathId, blobContent, idRow, pathRow)
```

Reply:
yes, looks much better, thanks